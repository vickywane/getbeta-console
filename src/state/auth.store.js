import { action, observable, decorate } from "mobx"

class AuthStore {
  authenticated = true
  authState = "Login"

  AuthUser = details => {
    const { id, name } = details

    localStorage.setItem("user_id", id)
    localStorage.setItem("user_name", name)
    localStorage.setItem("welcomeModal", true)

    this.authenticated = true
  }

  Login = () => {
    this.authenticated = true
  }

  LogOut = () => {
    localStorage.clear()

    this.authenticated = false
  }

  // App user's state . =====>
  // might be made into a sperate store .
  //mocked values to work with UI
  hasEvent = false
  hasVolunteer = false

  setEvent = () => {
    this.hasEvent = true
  }
  setVolunter = () => {
    this.hasVolunteer = true
  }

  setAuthState = state => {
    this.authState = state
  }
}

const DecoratedAuthStore = decorate(AuthStore, {
  //observables
  authenticated: observable,
  hasEvent: observable,
  hasVolunteer: observable,
  authState: observable,

  //actions
  Login: action,
  AuthUser: action,
  LogOut: action,
  setEvent: action,
  setVolunter: action,
  setAuthState: action,
})

const store = new DecoratedAuthStore()

export default store
