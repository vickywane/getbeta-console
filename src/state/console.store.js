import { action, observable, decorate } from "mobx";

class ConsoleStore {
  ProfilePane = false;

  showProfilePane = () => {
    this.ProfilePane = true;
  };

  closeProfilePane = () => {
    this.ProfilePane = false;
  };
}

const DecoratedConsoleStore = decorate(ConsoleStore, {
  //observables
  ProfilePane: observable,

  //actions
  showProfilePane: action,
  closeProfilePane: action
});

const store = new DecoratedConsoleStore();

export default store;
