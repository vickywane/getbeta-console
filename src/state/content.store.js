import Axios from 'axios'
import { action, observable, decorate } from 'mobx'
import { navigate } from '@reach/router'
import * as Sentry from '@sentry/react'

const CONTENT_ENDPOINT = `${process.env.REACT_APP_PRODUCTION_API_URI}/vendors`
const SUBSCRIPTIONS_ENDPOINT = `${process.env.REACT_APP_PRODUCTION_API_URI}/vendors/subscriptions`

const id = localStorage.getItem('userId')
const token = localStorage.getItem('token')

class ContentStore {
  contents = [] // all contents
  content = [] // single content
  isCreatingContent = false
  isLoadingContents = false

  @observable contentSubscribers = []
  @observable isLoading = false

  fetchContents = () => {
    this.isLoadingContents = true
    Axios.get(`${CONTENT_ENDPOINT}/contents/find/all`, { headers: { 'x-auth-token': token } })
      .then(res => {
        this.isLoadingContents = false
        this.contents = res.data.contents
      })
      .catch(e => console.log(`error : ${e}`))
  }

  createContent = (contentName, contentDescription, contentPrice, contentType, contentImage) => {
    this.isLoading = true

    Axios.post(
      `${CONTENT_ENDPOINT}/${id}/addContent`,
      {
        title: contentName,
        descrp: contentDescription,
        price: contentPrice,
        type: contentType,
        file: contentImage
      },
      {
        headers: {
          'x-auth-token': token
        }
      }
    )
      .then(result => {
        if (contentImage) {
          const { _id } = result.data.content

          const formData = new FormData()
          formData.append('file', contentImage)

          Axios.put(`${CONTENT_ENDPOINT}/${id}/${_id}/img`, formData, {
            headers: { 'x-auth-token': token, 'Content-Type': 'mulipart/formdata' }
          })
            .then(() => {
              this.isLoading = false

              navigate('/contents')
            })
            .catch(e => Sentry.captureException(e))
        }

        this.isLoading = false
        navigate('/contents')
      })
      .catch(e => {
        this.isLoading = false
        Sentry.captureException(e)
      })
  }

  updateContent = (contentId, title, description) => {
    this.isLoading = true
    Axios.post(
      `${CONTENT_ENDPOINT}/${id}/${contentId}/update`,
      { title, description },
      { headers: { 'x-auth-token': token } }
    )
      .then(response => {
        this.isLoading = false

        this.content = response.data.content
      })
      .catch(e => {
        this.isLoading = false

        Sentry.captureException(e)
      })
  }

  @action subscribeToContent = userId => {}

  @observable
  isCreatingContentFile = false

  @action userSubscribedContent = userId => {
    this.isLoading = true
    Axios.get(`${SUBSCRIPTIONS_ENDPOINT}/${userId}/subscribed-courses`)
      .then(res => {
        this.isLoading = false
      })
      .catch(e => {
        this.isLoading = false
        Sentry.captureException(e)
      })
  }

  @action
  addContentFile = (id, contentFile) => {
    this.isLoading = true
    this.isCreatingContentFile = true
    const contentfile = new FormData()
    contentfile.append('file', contentFile)

    Axios.post(`${CONTENT_ENDPOINT}/content/${id}/addfile`, contentfile, {
      headers: { 'x-auth-token': token, 'Content-Type': 'multipart/formdata' }
    })
      .then(res => {
        this.isCreatingContentFile = false
        this.isLoading = false
      })
      .catch(e => {
        this.isLoading = false
        Sentry.captureException(e)
      })
  }

  @observable contentFiles = []
  @action getContentFiles = id => {
    this.isLoadingContents = true
    Axios.get(`${CONTENT_ENDPOINT}/contents/${id}/find-files`, {
      headers: {
        'x-auth-token': token
      }
    })
      .then(res => {
        this.isLoadingContents = false
        this.contentFiles = res.data.contentfiles
      })
      .catch(e => {
        Sentry.captureException(e)
        this.isLoadingContents = false
      })
  }

  getUserContents = id => {
    this.isLoadingContents = true
    Axios.get(`${CONTENT_ENDPOINT}/${localStorage.getItem('userId')}/contents/find`, {
      headers: {
        'x-auth-token': localStorage.getItem('token')
      }
    })
      .then(res => {
        this.isLoadingContents = false

        this.contents = res.data.contents
      })
      .catch(e => {
        this.isLoadingContents = false

        Sentry.captureException(e)
      })
  }

  getContent = contentId => {
    this.isLoadingContents = true
    Axios.get(`${CONTENT_ENDPOINT}/${id}/${contentId}/find`, {
      headers: {
        'x-auth-token': token
      }
    })
      .then(res => {
        this.isLoadingContents = false

        this.content = res.data.content
      })
      .catch(e => {
        this.isLoadingContents = false

        Sentry.captureException(e)
      })
  }

  deleteContent = id => {
    Axios.delete(`${CONTENT_ENDPOINT}/${localStorage.getItem('userId')}/${id}/delete`, {
      headers: { 'x-auth-token': token }
    })
      .then(res => console.log(res))
      .catch(e => Sentry.captureException(e))
  }

  @observable
  contentDeleted = false

  @action
  deleteContentFile = contentId => {
    console.log(contentId, 'id')
    this.isLoading = true

    Axios.delete(`${CONTENT_ENDPOINT}/${contentId}/delete-file`, {
      headers: {
        'x-auth-token': token
      }
    })
      .then(() => {
        this.contentDeleted = true

        setTimeout(() => (this.contentDeleted = false), 3000)
      })
      .catch(e => {
        Sentry.captureException(e)
        this.isLoading = false
      })
  }
}

const DecoratedContentStore = decorate(ContentStore, {
  //observables
  contents: observable,
  content: observable,
  isCreatingContent: observable,
  isLoadingContents: observable,

  //actions
  getContent: action,
  fetchContents: action,
  getUserContents: action,
  createContent: action,
  updateContent: action,
  deleteContent: action
})

const store = new DecoratedContentStore()

export default store
