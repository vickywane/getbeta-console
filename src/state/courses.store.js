import Axios from 'axios'
import { action, observable, decorate } from 'mobx'
import { navigate } from '@reach/router'

const COURSE_ENDPOINT = `${process.env.REACT_APP_API_URL}/courses`
const token = localStorage.getItem('token')

class CourseStore {
  courses = []

  fetchCourse = id => {}

  fetchCourses = () => {
    let resData = []

    Axios.get(`${COURSE_ENDPOINT}/getfiles`, { headers: { 'x-auth-token': token } })
      .then(res => {
        resData.push(res.data.files)
      })
      .catch(e => console.log(e))

    return resData
  }

  createCourse = () => {
    const id = localStorage.getItem('userId')

    Axios.post(`${COURSE_ENDPOINT}/${id}/add`, { data: {} }, { headers: { 'x-auth-token': token } })
      .then(res => console.log(res))
      .catch(e => console.log(e))
  }
}

const DecoratedCourseStore = decorate(CourseStore, {
  //observables
  courses: observable,

  //actions
  fetchCourse: action,
  createCourse: action,
  fetchCourses: action
})

const store = new DecoratedCourseStore()

export default store
