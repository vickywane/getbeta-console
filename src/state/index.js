// export {  store as UserStore } from './user.store'
export { default as CourseStore } from './courses.store'
export { default as VendorStore } from './vendor.store'
export { default as ContentStore } from './content.store'
