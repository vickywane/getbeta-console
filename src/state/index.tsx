export { default as ConsoleStore } from "./console.store";
export { default as MediaStore } from "./media.store";
export { default as AuthStore } from "./auth.store";
