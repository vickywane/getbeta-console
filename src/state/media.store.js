import { action, observable, decorate } from "mobx"

class MediaStore {
  Empty = true

  UploadModal = false

  openUploadModal = () => {
    this.UploadModal = true
  }

  closeUploadModal = () => {
    this.UploadModal = false
  }

  items = [
    {
      id: 1,
      name: "",
      url: "",
    },
  ]

  itemsNo = null

  AddItem = () => {
    this.empty = false
  }

  DeleteItem = () => {}
}

const DecoratedMediaStore = decorate(MediaStore, {
  //observables
  Empty: observable,
  itemsNo: observable,
  items: observable,
  UploadModal: observable,

  //actions
  AddItems: action,
  DeleteItems: action,

  openUploadModal: action,
  closeUploadModal: action,
})

const store = new DecoratedMediaStore()

export default store
