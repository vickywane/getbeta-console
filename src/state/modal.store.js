import { action, observable, decorate } from "mobx"

class ModalStore {
  // checklist modal
  showChecklist = false

  openChecklist = () => {
    this.showChecklist = true
  }

  closeChecklist = () => {
    this.showChecklist = false
  }

  // people modal
  showPeople = false
  invite = false
  search = false

  openPeople = () => {
    this.showPeople = true
  }

  closePeople = () => {
    this.showPeople = false
  }

  beginInvite = () => {
    this.invite = true
  }

  beginSearch = () => {
    this.search = true
  }

  stopSearch = () => {
    this.search = false
  }

  //contact modal
  showContactModal = false

  openContactModal = () => {
    this.showContactModal = true
  }

  closeContactModal = () => {
    this.showContactModal = false
  }

  // Forms Modal
  showFormModal = false

  openFormModal = () => {
    this.showFormModal = true
  }

  closeFormModal = () => {
    this.showFormModal = false
  }

  // welcome modal
  showWelcomeModal = true

  closeWelcomeModal = () => {
    this.showWelcomeModal = false
  }
}

const DecoratedModalStore = decorate(ModalStore, {
  //observables

  // checklist ------------>
  showChecklist: observable,

  // people ======>
  showPeople: observable,
  invite: observable,
  search: observable,

  //actions
  // checklist ------------>
  openChecklist: action,
  closeChecklist: action,

  openPeople: action,
  closePeople: action,
  beginInvite: action,
  beginSearch: action,
  stopSearch: action,

  // contact
  showContactModal: observable,

  openContactModal: action,
  closeContactModal: action,

  // forms
  showFormModal: observable,

  openFormModal: action,
  closeFormModal: action,

  showWelcomeModal: observable,
  closeWelcomeModal: action,
})

const store = new DecoratedModalStore()

export default store
