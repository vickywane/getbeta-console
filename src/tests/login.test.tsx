import React from "react"
import Enzyme, { shallow } from "enzyme"
import Adapter from "enzyme-adapter-react-16"

import Login from "../pages/auth/login/signin"
import { AuthStore } from "../state/"

Enzyme.configure({ adapter: new Adapter() })

describe("Tests for the Login Page", () => {
  const Component = shallow(<Login AuthStore={AuthStore} />)

  it("Login Page is rendered with specific Elements", () => {
    expect(Component.find("h4"))

    expect(Component.exists(".input"))
    // expect(Component.find("input")).toHaveLength(6)
  })
})
