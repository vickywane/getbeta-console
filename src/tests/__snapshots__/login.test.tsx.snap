// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the Login Page Login Page is rendered with specific Elements 1`] = `ShallowWrapper {}`;
